libcontainer/container_linux.go

```diff
@@ -2164,11 +2162,16 @@ func (c *Container) bootstrapData(cloneFlags uintptr, nsMaps map[configs.Namespa
 	if !joinExistingUser {
 		// write uid mappings
 		if len(c.config.UidMappings) > 0 {
-			if c.config.RootlessEUID && c.newuidmapPath != "" {
-				r.AddData(&Bytemsg{
-					Type:  UidmapPathAttr,
-					Value: []byte(c.newuidmapPath),
-				})
+			if c.config.RootlessEUID {
+				// We resolve the paths for new{u,g}idmap from
+				// the context of runc to avoid doing a path
+				// lookup in the nsexec context.
+				if path, err := exec.LookPath("newuidmap"); err == nil {
```

Comment:
Silly question; I haven't fully checked which codepaths are executed, but I see that `bootstrapData()` is called multiple times; can this lead to `exec.LookPath()` being called multiple times (which will be iterating over all directories in `PATH`); is that a concern? https://github.com/golang/go/blob/f229e7031a6efb2f23241b5da000c3b3203081d6/src/os/exec/lp_unix.go#L47-L57

Reply:
bootstrapData is caller either for `runc start`/`run`, or for `runc init`